technology/kotlin-notebook/school-timetabling.ipynb

```diff
@@ -0,0 +1,641 @@
+      "| TUE 09:30  | Math       |            | Physics    |\n",
+      "|            | A. Turing  |            | M. Curie   |\n",
+      "|            | 10th grade |            | 9th grade  |\n",
+      "|------------|------------|------------|------------|\n",
+      "| TUE 10:30  | Math       |            | Chemistry  |\n",
+      "|            | A. Turing  |            | M. Curie   |\n",
+      "|            | 9th grade  |            | 10th grade |\n",
+      "|------------|------------|------------|------------|\n",
+      "| TUE 13:30  |            | English    | Physics    |\n",
+      "|            |            | I. Jones   | M. Curie   |\n",
+      "|            |            | 9th grade  | 10th grade |\n",
+      "|------------|------------|------------|------------|\n",
+      "| TUE 14:30  |            | History    | French     |\n",
+      "|            |            | I. Jones   | M. Curie   |\n",
+      "|            |            | 9th grade  | 10th grade |\n",
+      "|------------|------------|------------|------------|\n"
+     ]
+    }
+   ],
+   "source": [
+    "import ai.timefold.solver.core.config.solver.SolverConfig\n",
+    "import ai.timefold.solver.core.api.solver.SolverFactory\n",
+    "import ai.timefold.solver.core.api.solver.Solver\n",
+    "import ai.timefold.solver.core.api.solver.SolutionManager\n",
+    "import ai.timefold.solver.core.api.solver.SolverManager\n",
+    "\n",
+    "val solverFactory: SolverFactory<TimeTable> = SolverFactory.create(SolverConfig()\n",
+    "        .withSolutionClass(TimeTable::class.java)\n",
+    "        .withEntityClasses(Lesson::class.java)\n",
+    "        .withConstraintProviderClass(TimeTableConstraintProvider::class.java)\n",
+    "        // The solver runs only for 5 seconds on this small dataset.\n",
+    "        // It's recommended to run for at least 5 minutes (\"5m\") otherwise.\n",
+    "        .withTerminationSpentLimit(Duration.ofSeconds(5)))\n",
+    "\n",
+    "println(\"Loading the problem ...\")\n",
+    "val problem: TimeTable = generateDemoData()\n",
+    "\n",
+    "println(\"Solving the problem ...\")\n",
+    "val solver: Solver<TimeTable> = solverFactory.buildSolver()\n",
+    "val solution: TimeTable = solver.solve(problem)\n",
+    "\n",
+    "println(\"Visualizing the solution ...\")\n",
+    "printTimeTable(solution)\n"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2023-10-30T21:01:08.485939578Z",
+     "start_time": "2023-10-30T21:01:03.343882883Z"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "To learn more, visit [timefold.ai](https://timefold.ai)."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "Kotlin",
+   "language": "kotlin",
+   "name": "kotlin"
+  },
+  "language_info": {
+   "name": "kotlin",
+   "version": "1.9.0",
+   "mimetype": "text/x-kotlin",
```

Comment:
How do we update this? Typically our kotlin version is bumped by Dependabot, but I doubt it will see this.